src/main/webapp/js/lib/calendar.js

```diff
@@ -206,6 +249,22 @@ function createDateFromTimeString(time, day) {
  */
 function clear() {
   calendar.clear(/*immediately=*/ true);
+  // Duplicate the ORIGINAL_COLORS array and rerandomize the potential schedule
+  // colors.
+  scheduleColors = [...ORIGINAL_COLORS];
+  randomizeColors();
+}
+
+/**
+ * Randomizes the scheduleColors array.
+ */
+function randomizeColors() {
+  for (let i = scheduleColors.length - 1; i > 0; i--) {
+    const j = Math.floor(Math.random() * i);
+    const temp = scheduleColors[i];
+    scheduleColors[i] = scheduleColors[j];
+    scheduleColors[j] = temp;
+  }
 }
 
```

Comment:
it may be better to not randomize this array -- because we have pagination now, it may be better to stay consistent with colors between schedules (meaning a consistent order before and after clearing the schedule).

Reply:
Maybe randomize it once and then never again? Otherwise every schedule would be rainbow colored starting from red, orange, yellow, etc.

Reply:
I think that's a good idea